src/Analyzer.BicepProcessor/BicepTemplateProcessor.cs

```diff
@@ -91,47 +98,42 @@ public static (string, BicepMetadata) ConvertBicepToJson(string bicepPath)
                 throw new Exception($"Bicep issues found:{SysEnvironment.NewLine}{string.Join(SysEnvironment.NewLine, bicepIssues)}");
             }
 
-            string GetPathRelativeToEntryPoint(string absolutePath) => Path.GetRelativePath(
-                Path.GetDirectoryName(compilation.SourceFileGrouping.EntryPoint.FileUri.AbsolutePath), absolutePath);
+            string entryPointDirectory = Path.GetDirectoryName(compilation.SourceFileGrouping.EntryPoint.Uri.AbsolutePath);
+
+            bool IsLocalModuleReference(KeyValuePair<IArtifactReferenceSyntax, ArtifactResolutionInfo> artifact) =>
+                artifact.Key is ModuleDeclarationSyntax moduleDeclaration &&
+                moduleDeclaration.Path is StringSyntax moduleDeclarationPath &&
+                !moduleDeclarationPath.SegmentValues.Any(IsModuleRegistryPathRegex.IsMatch) &&
+                artifact.Value.Result.IsSuccess();
 
             // Collect all needed module info from SourceFileGrouping metadata
-            var moduleInfo = compilation.SourceFileGrouping.FileUriResultByArtifactReference.Select(sourceFileAndMetadata =>
-            {
-                var bicepSourceFile = sourceFileAndMetadata.Key as BicepSourceFile;
-                var pathRelativeToEntryPoint = GetPathRelativeToEntryPoint(bicepSourceFile.FileUri.AbsolutePath);
-                var modules = sourceFileAndMetadata.Value
-                    .Select(artifactRefAndUriResult =>
+            var moduleInfo = compilation.SourceFileGrouping.ArtifactLookup
+                .Where(IsLocalModuleReference)
+                .GroupBy(artifact => artifact.Value.Origin)
+                .Select(grouping =>
+                {
+                    var bicepSourceFile = grouping.Key;
+                    var pathRelativeToEntryPoint = Path.GetRelativePath(
+                        Path.GetDirectoryName(compilation.SourceFileGrouping.EntryPoint.Uri.AbsolutePath), bicepSourceFile.Uri.AbsolutePath);
+
+                    var modules = grouping.Select(artifactRefAndUriResult =>
                     {
-                        // Do not include modules imported from public/private registries, as it is more useful for user to see line number
```

Comment:
maybe keep this comment. Seems similar logic is kept

Reply:
Added more comments too.